Collector/ProfileClient.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+use Http\Client\Common\FlexibleHttpClient;
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * The ProfileClient decorates any client that implement both HttpClient and HttpAsyncClient interfaces to gather target
+ * url and response status code.
+ *
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ *
+ * @internal
+ */
+class ProfileClient implements HttpClient, HttpAsyncClient
+{
+    /**
+     * @var HttpClient|HttpAsyncClient
+     */
+    private $client;
+
+    /**
+     * @var Collector
+     */
+    private $collector;
+
+    /**
+     * @param HttpClient|HttpAsyncClient $client    The client to profile. Client must implement both HttpClient and
```

Comment:
The ISP states not to make big interfaces, so that people don't need to implement all of them. In your case, they still can choose to implement either inherited interface, so ISP wouldn't be violated here IMO. And if you do need both methods, it looks like the best bay to express that need.